dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java

```diff
@@ -668,4 +659,81 @@ private UrlMapContext getUrlMapContext(final User systemUser, final Host host, f
     private UrlMapContext getUrlMapContext(final User systemUser, final Host host, final String uri) {
         return getUrlMapContext(systemUser, host, uri, PageMode.PREVIEW_MODE);
     }
+
+    /**
+     * methodToTest {@link URLMapAPIImpl#processURLMap(UrlMapContext)}
+     * Given Scenario: Process a URL Map url when both the Content Type and Content exists but the field to Map is a
+     * {@link com.dotcms.contenttype.model.field.DataTypes#INTEGER}
+     * ExpectedResult: Should return a {@link URLMapInfo} wit the right content ans detail page
+     */
+    @Test
+    public void shouldReturnContentletWhenTheContentExistsAndUseAIntegerFIeld()
+            throws DotDataException, DotSecurityException {
+        final String newsPatternPrefix =
+                "integer" + TEST_PATTERN + System.currentTimeMillis() + "/";
+
+
+        final Field field = new FieldDataGen().dataType(DataTypes.INTEGER).next();
+        final String urlMapper = newsPatternPrefix  + "{" + field.variable() + "}";
+        final ContentType contentType = new ContentTypeDataGen()
+                .field(field)
+                .detailPage(detailPage1.getIdentifier())
+                .urlMapPattern(urlMapper)
+                .nextPersisted();
+
+        final Contentlet newsTestContent = new ContentletDataGen(contentType.id())
+                .setProperty(field.variable(), 2)
+                .languageId(1)
+                .host(host)
+                .nextPersisted();
+
+        final UrlMapContext context = getUrlMapContext(systemUser, host,
+                newsPatternPrefix + newsTestContent.getStringProperty(field.variable()));
+
+        final Optional<URLMapInfo> urlMapInfoOptional = urlMapAPI.processURLMap(context);
+
+        assertTrue(urlMapInfoOptional.isPresent());
+        final URLMapInfo urlMapInfo = urlMapInfoOptional.get();
+        assertEquals(newsTestContent.getLongProperty(field.variable()),
+                urlMapInfo.getContentlet().getLongProperty(field.variable()));
+        assertEquals("/news-events/news/news-detail", urlMapInfo.getIdentifier().getURI());
+    }
+
+    /**
+     * methodToTest {@link URLMapAPIImpl#processURLMap(UrlMapContext)}
+     * Given Scenario: Process a URL Map url when both the Content Type and Content exists but the field to Map is a
+     * {@link com.dotcms.contenttype.model.field.DataTypes#INTEGER}
```

Comment:
Update this comment

Reply:
done https://github.com/dotCMS/core/pull/19459/commits/f17437f7adc0402fb60bbb10ce4d32760dd8a60d#diff-de0fb9193432881efb85f6e38c30df1524a2a3e392f54d744734dd37520586daR705